zrml/swaps/src/benchmarks.rs

```diff
@@ -560,8 +558,8 @@ benchmarks! {
         .saturated_into();
         let weight_out = T::MinWeight::get();
         let weight_in = 4 * weight_out;
-        let mut weights = vec![weight_in; asset_count as usize];
-        weights[asset_count as usize - 1] = weight_out;
+        let mut weights = vec![weight_out; asset_count as usize];
+        weights[0] = weight_in;
```

Comment:
This change seems complicated. We have now `asset_count - 1` times `MinWeight` and for the first weight `4 * MinWeight`. Instead of `asset_count - 1` times `4 * MinWeight` and for the last weight `MinWeight`. Can you elaborate on that a little bit?

Reply:
Sure. This test raises `MaxTotalWeight`, which means that the total weight (which is the sum of all weights) is too large. So we need to make some weights smaller.

Previously, the weight vector was initialized as `[4, ..., 4]` and then the last weight, which is the weight of the outgoing asset, was set to `1`. This results in a large weight. But we only need the first (ingoing) and last (outgoing) asset for this test, so we don't care about all the weights in the middle.

Now we initialize the vector as `[1, ..., 1]` and only change the first weight (of the ingoing asset) to `4`, which results in a much lower weight, which, fortunately, is smaller than the maximum total weight.